components/butterfly/src/member.rs

```diff
@@ -678,58 +731,67 @@ impl MemberList {
         }
     }
 
-    /// Iterates over every suspected membership entry, calling the given closure.
-    pub fn with_suspects<F>(&self, mut with_closure: F) -> ()
-    where
-        F: FnMut((&str, &SteadyTime)) -> (),
-    {
-        for (id, suspect) in self
-            .suspect
-            .read()
-            .expect("Suspect list lock is poisoned")
-            .iter()
+    // TODO (CM): Better return type than Vec<String>
+    /// Query the list of aging Suspect members to find those which have now
+    /// timed out to Confirmed. Health is updated appropriately, and a
+    /// list of newly-Confirmed Member IDs is returned.
+    pub fn get_newly_expired_confirmed(&self, timeout: Duration) -> Vec<String> {
+        let now = SteadyTime::now();
+        let mut newly_confirmed: Vec<String> = Vec::new();
+
         {
-            with_closure((id, suspect));
+            let mut aging_suspects = self
```

Comment:
I don't think this needs to be in a separate block; it seems to build without it for me. Also, you can do without the `aging_suspects` variable entirely:
```rust
        self.aging_suspects
            .write()
            .expect("aging_suspects lock is poisoned")
            .retain(|ref member_id, ref suspect_time| {
                let timed_out = now >= **suspect_time + timeout;
                if timed_out {
                    newly_confirmed.push(member_id.to_string());
                }
                !timed_out
            });
```
Also, and this is more my personal style of what I find easier to grok, you could eliminate the `timed_out` variable and write the inner block as:
```rust
                if now >= **suspect_time + timeout {
                    newly_confirmed.push(member_id.to_string());
                    false
                } else {
                    true
                }
```
This is mostly to eliminate the `!timed_out` return value. For me interpreting a negated return to `retain` is somewhat trickier than just seeing the two branches with explicit `true` and `false` returns. However this is more of a style thing, so take or leave it.

Reply:
Good calls; looks like I was unnecessarily retaining some structure that was needed in prior versions of this refactoring.

I'll tweak the booleans, too. For some reason, the `timed_out` formulation read better to me initially, but now they're both fine. I was probably using this as a way to more thoroughly figure out / document what the code was doing during the various refactorings that led here. Now it's clear enough without `timed_out`, I think.